store/postgres/src/entities.rs

```diff
@@ -0,0 +1,993 @@
+            ValueType::Boolean
+            | ValueType::BigInt
+            | ValueType::Bytes
+            | ValueType::BigDecimal
+            | ValueType::ID
+            | ValueType::Int => (String::from("btree"), String::from(""), "->>"),
+            ValueType::String => (String::from("gin"), String::from("gin_trgm_ops"), "->>"),
+            ValueType::List => (String::from("gin"), String::from("jsonb_path_ops"), "->"),
+        };
+
+        // It is not to be possible to use bind variables in this code,
+        // and we have to interpolate everything into the query directly.
+        // We also have to use conn.batch_execute to issue the `create index`
+        // commands; using `sql_query(..).execute(conn)` will make the database
+        // accept the commands, and log them as if they were successful, but
+        // without any effect on the schema
+        match self {
+            Table::Public(_) => {
+                let query = format!("
+                    create index if not exists {name} on public.entities 
+                    using {index_type} ((data->'{attribute_name}'{jsonb_operator}'data') {index_operator})
+                    where subgraph='{subgraph}' and entity='{entity_name}'",
+                    name=&index.index_name,
+                    index_type=index_type,
+                    attribute_name=&index.attribute_name,
+                    jsonb_operator=jsonb_operator,
+                    index_operator=index_operator,
+                    subgraph=index.subgraph_id.to_string(),
+                    entity_name=&index.entity_name);
+                conn.batch_execute(&*query)?;
+                Ok(1)
+            }
+            Table::Split(entities) => {
+                let name = format!(
+                    "entities_{}_{}",
+                    to_snake_case(&index.entity_name),
+                    to_snake_case(&index.attribute_name)
+                );
+                let query = format!("
+                    create index if not exists {name} on {subgraph}.entities 
+                    using {index_type} ((data->'{attribute_name}'{jsonb_operator}'data') {index_operator})
+                    where entity='{entity_name}'",
+                    name=name,
+                    subgraph=entities.schema,
+                    index_type=index_type,
+                    attribute_name=&index.attribute_name,
+                    jsonb_operator=jsonb_operator,
+                    index_operator=index_operator,
+                    entity_name=&index.entity_name);
+                conn.batch_execute(&*query)?;
+                Ok(1)
+            }
+        }
+    }
+}
+
+/// Delete all entities. This function exists solely for integration tests
+/// and should never be called from any other code. Unfortunately, Rust makes
+/// it very hard to export items just for testing
+pub fn delete_all_entities(conn: &PgConnection) -> Result<usize, StoreError> {
```

Comment:
To make it even more obvious that this is dangerous, we could put 'dangerous' or similar into the name of the function.

Reply:
A trick I've used is to stick `#[cfg(debug)]` on top of test-only builds, this way it can never make it into the release builds we use for prod.

Reply:
I renamed it to `delete_all_entities_for_test_use_only`

Reply:
@leodasvacas just saw your comment about `#[cfg(debug)]` now. When I stick that into the code, the store tests fail to compile; when I also annotate where this gets called in `remove_test_data` in the store tests, I get a warning about `conn` not being used in VSCode (but not release builds strangely). I'll leave that out therefore, and just stick with the scary name.

Reply:
@lutter sorry my suggestion was wrong, the correct attribute is `#[cfg(debug_assertions)]`.